django/core/management/commands/sqlmigrate.py

```diff
@@ -53,15 +53,20 @@ def handle(self, *args, **options):
         except KeyError:
             raise CommandError("Cannot find a migration matching '%s' from app '%s'. Is it in INSTALLED_APPS?" % (
                 migration_name, app_label))
-        targets = [(app_label, migration.name)]
+        target = (app_label, migration.name)
 
         # Show begin/end around output for atomic migrations, if the database
         # supports transactional DDL.
         self.output_transaction = migration.atomic and connection.features.can_rollback_ddl
 
+        # If migration was replaced, load the graph without replacements
+        # in order to show SQL plan
+        if target in executor.loader.graph.replaced_node_keys:
+            executor = MigrationExecutor(connection, replace_migrations=False)
```

Comment:
Won't it be faster to just make the inital load not use replacement at all now?

Reply:
> Won't it be faster to just make the initial load not use replacement at all now?

It seems to also allow inspecting replacement/squash migrations when not loading them :+1: Seems logical since we still load them because they are in the disk_migrations, but we just don't remove the replaced migrations. Good call, thanks!